gallery/visualtests/nxDrawer.js

```diff
@@ -0,0 +1,168 @@
+
+    const [openDrawerBtn] = await waitAndGetElements(openDrawerBtnSelector);
+
+    await openDrawerBtn.click();
+
+    await wait(200);
+
+    const [drawer] = await waitAndGetElements(`#${drawerId}`);
+
+    return drawer;
+  }
+
+  function simpleDrawerTest(drawerId, buttonId) {
+    return async function() {
+      const drawer = await openDrawer(drawerId, buttonId);
+
+      await checkScreenshot(drawer);
+    };
+  }
+
+  function drawerA11yTest(drawerId, buttonId) {
+    return async function() {
+      await openDrawer(drawerId, buttonId);
+      await a11yTest();
+    };
+  }
+
+  beforeEach(async function() {
+    await getPage().setViewport({ width: 1366, height: 1000 });
+  });
+
+  describe('Simple NxDrawer', function() {
+    it('looks right', simpleDrawerTest('nx-drawer-simple', 'nx-drawer-simple-open-button'));
+
+    it('passes a11y checks', drawerA11yTest('nx-drawer-simple', 'nx-drawer-simple-open-button'));
+  });
+
+  describe('Narrow NxDrawer', function() {
+    it('looks right', simpleDrawerTest('nx-drawer-narrow', 'nx-drawer-narrow-open-button'));
+  });
+
+  describe('NxDrawer With Footer', function() {
+    it('looks right', simpleDrawerTest('nx-drawer-with-footer', 'nx-drawer-with-footer-open-button'));
+  });
+
+  describe('NxDrawer With Subtitle', function() {
+    it('looks right', simpleDrawerTest('nx-drawer-with-subtitle', 'nx-drawer-with-subtitle-open-button'));
+  });
+
+  describe('NxDrawer With Paragraph', function() {
+    it('looks right', simpleDrawerTest('nx-drawer-with-paragraph', 'nx-drawer-with-paragraph-open-button'));
+  });
+
+  describe('NxDrawer With Subtitle and Paragraph', function() {
+    it('looks right', simpleDrawerTest('nx-drawer-with-subtitle-and-paragraph',
+        'nx-drawer-with-subtitle-and-paragraph-open-button'));
+  });
+
+  describe('NxDrawer + NxDropdown ESC Closing Behavior', function() {
+    async function isFocused(el) {
+      return el.evaluate(e => e === document.activeElement);
+    }
+
+    async function isInDocument(el) {
+      return el.evaluate(e => e.isConnected);
+    }
+
+    async function pressEsc() {
+      const { keyboard } = getPage();
+
+      await keyboard.press('Escape');
+    }
+
+    it('closes one layer per ESC press', async function() {
+      const [openButton] = await waitAndGetElements('#nx-drawer-esc-open-button');
+
+      await openButton.click();
+
+      await wait(200);
+
+      const [drawer, dropdownToggle] = await waitAndGetElements(
+          '#nx-drawer-esc',
+          '#nx-drawer-esc .nx-dropdown__toggle'
+      );
+
+      expect(await isInDocument(drawer)).toBe(true);
+      expect(await isInDocument(dropdownToggle)).toBe(true);
+
+      await dropdownToggle.click();
+
+      const [dropdownMenu] = await waitAndGetElements('#nx-drawer-esc .nx-dropdown-menu');
+
+      expect(await isInDocument(dropdownMenu)).toBe(true);
+
+      await pressEsc();
+
+      expect(await isInDocument(dropdownMenu)).toBe(false);
```

Comment:
Isn't the crucial part of the test that `drawer` _is_ still in the document at this point? We should test that.  Also check where focus is at this point.  I assume that should be the dropdown toggle.

Reply:
> test that

Fixed here: 8356f233f575973b8be5be477b77d35da0b492c6